docs/BOOT_LOADER_SPECIFICATION.md

```diff
@@ -393,6 +388,85 @@ creating a partition and file system for it) and creates the `/loader/entries/`
 directory in it. It then installs an appropriate boot loader that can read
 these snippets. Finally, it installs one or more kernel packages.
 
+## Sorting
+
+The boot loader menu should generally show entries in some order meaningful to
+the user. The `title` key is free-form and not suitable to be used as the
+primary sorting key. Instead, in order of priority, the `sort-key`
+(lexicographically increasing), `machine-id` (lexicographically increasing),
```

Comment:
User interfaces these days say just `A-Z`/`Z-A` instead of complicated words such as (lexicographically) ascending. Imho, this would much easier to grok.

Reply:
Right, but the thing is that this is not just an alphanumerical sort, but that the string has "structure". So I can't say that comparsion 9-0 is used for digits, because it's more like "skip 0, but not if it's the only thing, and then evaluate as number, order +∞ - 0".

I added "A-Z" or "Z-A" in the description, but didn't remove the existing text.

Reply:
Hmm, but now I realize that the text is wrong in one important aspect: it implies that strverscmp can be used for all keys, including e.g. machine-id. But a normal alphabetical sort should be used there.